runtime/domain.c

```diff
@@ -1146,114 +1147,148 @@ CAMLexport _Atomic caml_timing_hook caml_domain_terminated_hook =
 
 static void domain_terminate(void);
 
-static value make_finished(caml_result result)
+static value make_finished(value res_or_exn)
 {
   CAMLparam0();
   CAMLlocal1(res);
-  res = caml_alloc_1(
-    (caml_result_is_exception(result) ?
-     1 /* Error */ :
-     0 /* Ok */),
-    result.data);
+  if (Is_exception_result(res_or_exn)) {
+    res = Extract_exception(res_or_exn);
+    /* [Error res] */
+    res = caml_alloc_1(1, res);
+  } else {
+    /* [Ok res_of_exn] */
```

Comment:
Did you mean
```suggestion
    /* [Ok res_or_exn] */
```
?

Reply:
Thanks! (that's actually a preexisting typo in temporarily reverted code)